pageserver/src/tenant.rs

```diff
@@ -2331,12 +2177,15 @@ impl Tenant {
         ancestor: Option<Arc<Timeline>>,
         remote_client: Option<RemoteTimelineClient>,
         init_order: Option<&InitializationOrder>,
+        validate_ancestor: bool,
```

Comment:
Cannot suggest, but would add a line to method docs as this ancestor timeline validation always confuses me, but now with an added twist.

```
///
/// `validate_ancestor == false` is used when a timeline is created for deletion
/// and we might not have the ancestor present anymore which is fine for to be
/// deleted timelines.
```

Additionally it is weird that we only do `anyhow::ensure!(new_metadata.ancestor_timeline.is_some() == ancestor.is_some())` compared to `anyhow::ensure!(new_metadata.ancestor_timeline == ancestor.as_ref().map(|t| t.timeline_id))`.

Reply:
Applied your suggestion. `validate_ancestor`  is a crutch, not sure there is non invasive way to implement a better solution